File: Udemy_2020/Regression/Simple Linear Regression/simple_linear_regression.r
#import dataset
Dataset = read.csv('Salary_Data.csv')

#splitting the testing and training data set
library(caTools)
set.seed(123)
split = sample.split(Dataset$Salary,SplitRatio = 2/3)
trainingset = subset(Dataset, split == TRUE)
testingset = subset(Dataset, split == FALSE)

#fitting the regressor for training set
regressor = lm(formula = Salary ~ YearsExperience,
               data =  trainingset)

#predcition

Y_pred = predict(regressor,newdata = testingset)

#Visualisations
#install.packages("ggplot2")
library(ggplot2)
ggplot() +
  geom_point(aes(x = trainingset$YearsExperience, y = trainingset$Salary),
             colour = 'red') +
  geom_line(aes(x = trainingset$YearsExperience, y = predict(regressor,newdata = trainingset)),
                colour = 'blue') +
  xlab("Years of Experience") +
  ylab("Salary")+
  ggtitle("Plot Training set")

#Visualisaton testing set
library(ggplot2)
ggplot() +
  geom_point(aes(x = testingset$YearsExperience, y = testingset$Salary),
             colour = 'red') +
  geom_line(aes(x = trainingset$YearsExperience,y = predict(regressor,newdata = trainingset)),
            colour = 'blue') +
  xlab("Years of Experience") +
  ylab("Salary")+
  ggtitle("Plot Testing set")

File: Udemy_2020/Regression/Simple Linear Regression/simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Wed Mar 18 00:11:29 2020

@author: keert
"""

#importing the libraries

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing the data sets
dataset = pd.read_csv('Salary_Data.csv')
#separating the dependent and the independent variables
#independent variables in the data set.
X = dataset.iloc[:,:-1].values
#dependentvariable vector
Y = dataset.iloc[:,1].values


#splitting test and training set
from sklearn.cross_validation import train_test_split
X_train,X_test,Y_train,Y_test = train_test_split(X,Y, test_size = 1/3, random_state = 0)

#simple linear regression model creation
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,Y_train)

#prediction of values
y_pred = regressor.predict(X_test)

#visualisation training set
plt.scatter(X_train,Y_train, color = "red")
plt.plot(X_train, regressor.predict(X_train), color = "blue")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.title("Training set")

#visualisation testing set
plt.scatter(X_test,Y_test, color = "red")
plt.plot(X_train,  regressor.predict(X_train), color = "blue")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.title("Testing set")